tests/scripts/analyze_outcomes.py

```diff
@@ -118,25 +118,24 @@ def analyze_driver_vs_reference(results: Results, outcomes,
     - only some specific test inside a test suite, for which the corresponding
       output string is provided
     """
-    available = check_test_cases.collect_available_test_cases()
-
-    for key in available:
-        # Continue if test was not executed by any component
-        hits = outcomes[key].hits() if key in outcomes else 0
-        if hits == 0:
-            continue
-
+    seen_reference_passing = False
+    for key in outcomes:
```

Comment:
The logic here is weird (and likely slower than it could be). Why iterate over all outcomes when you only want outcomes from two components? I think we should construct a table of outcomes indexed by components before doing the driver-related analyses. Probably at the same time as we construct the table indexed by test cases. The reason `outcomes` is indexed by test cases is that this is the only thing that was needed when this program was originally written.

I think here we want `Dict[str, Frozenset[Tuple[str, str]]]` mapping component names to the set of `(suite, description)` that pass.

This is partly preexisting and I could accept the argument that this is out of scope here.

Reply:
I think that's a very good idea, but indeed I'm going to argue it's out of scope here :)

Reply:
https://github.com/Mbed-TLS/mbedtls/issues/8423